### ISSUE TITLE
Hyperparameter grid search is extremely slow (joblib n_jobs=-1 + TF; also GPU disabled)

### ISSUE THREAD

**Issue by ayyucedemirbas** [bozdaglab/BioLM-NET#1]
Hello Professor Bozdag @serdarbozdag and @jubairmalik,

I am trying to run this project on Google Colab with T4 GPU. Running the hyperparameter grid in Colab takes forever. The current code uses joblib.Parallel(..., n_jobs=-1) to evaluate Keras models in parallel, and also sets os.environ['CUDA_VISIBLE_DEVICES'] = '' (which disables the GPU usage). Combined with recreating heavy mask/attention tensors inside each of the model creation and running epochs=200 with 5-fold CV, this produces very high CPU and memory overhead and near-infinite runtime in practice on Colab.

    `with tqdm_joblib(
            tqdm(desc="Evaluating hyperparameter combinations",
                 total=len(param_dicts))
         ) as progress_bar:
        results = Parallel(n_jobs=-1)(
            delayed(evaluate_combination)(
                param, X_gene_train, X_dna_train, y_train, n_splits=5
            )
            for param in param_dicts
        )`

I am going to try to solve this issue, however, I wanted to inform you too. I am very sorry for the disturbance. Thank you so much!

Kind Regards,
- Ayyuce Demirbas
